packages/legend-application-query/src/stores/QueryBuilderResultState.ts

```diff
@@ -172,7 +173,12 @@ export class QueryBuilderResultState {
     }
   }
 
-  *execute(): GeneratorFn<void> {
+  async execute(
```

Comment:
Generally, I agree with the approach, however, I think this logic should be put in the state instead. My suggestion is to create a new field in the reuslt state:

```ts
queryRunPromise: Promise<void> | undefined = undefined;
```

Also, let's rename this method to `runQuery` and the boolean flag to `isRunningQuery` instead of `isExecutingQuery`